docs(core): fix content errors

petebacondarwin opened the pull request:
This fixes all the "dgeni" errors for the `@angular/core` API docs.

petebacondarwin committed:
docs(core): fix content errors

petebacondarwin commented:
@IgorMinar I think this should be cherry picked to stable. Is that correct?

gkalpak commented on packages/core/src/metadata/ng_module.ts:
```diff
@@ -105,7 +103,7 @@ export interface NgModule {
    * Specifies a list of directives/pipes that belong to this module.
    *
-   * ### Example
+   * **Example**:
```
The `:` seems inconsistent with other similar changes (in other PRs :grin:)

gkalpak commented on packages/core/src/metadata/directives.ts:
```diff
@@ -205,7 +203,7 @@ export interface Directive {
    * Specify the events, actions, properties and attributes related to the host element.
    *
-   * ## Host Listeners
+   * **Host Listeners:**
```
`:`?

gkalpak commented on packages/core/src/metadata/directives.ts:
```diff
@@ -458,24 +450,21 @@ export interface ComponentDecorator {
    * * **viewProviders** - list of providers available to this component and its view children
    *
-   * ### Example
+   * **Example:**
```
You have to decide: Which side of `**` does the `:` go? 😛

petebacondarwin replied:
Perhaps I should have written a script ??

petebacondarwin commented on packages/core/src/application_ref.ts:
```diff
@@ -196,8 +196,7 @@ export class PlatformRef {
    *
-   * ## Simple Example
-   *
+   * ** The following example demonstrates calling `bootstrapModuleFactory`:**
```
😱

petebacondarwin committed:
fixup! docs(core): fix content errors

petebacondarwin committed:
fixup! docs(core): fix content errors

petebacondarwin committed:
build(aio): support `@example` tags in the API docs

petebacondarwin committed:
docs(core): fix up example headings and remove `{@example ..}` tags

The `{@example ... }` inline JSDOC tag is not really desirable anymore.
Instead it is preferable to use `<code-example>` tags instead.

mary-poppins commented:
You can preview bb5eeb2 at https://pr23666-bb5eeb2.ngbuilds.io/.
